src/lambda/vue/samInvokeVue.ts

```diff
@@ -109,61 +184,92 @@ export const Component = Vue.extend({
             },
             deep: true,
         },
-        payload: function (newval: string) {
-            this.resetJsonError()
+        payload: function (newval: {value: string, errorMsg: string}) {
+            this.resetJsonErrors()
             vscode.setState({
                 payload: newval,
                 launchConfig: this.launchConfig,
             })
         },
     },
     methods: {
-        resetJsonError() {
-            this.jsonError = ''
+        resetJsonErrors() {
+            this.payload.errorMsg = ''
+            this.environmentVariables.errorMsg = ''
+            this.headers.errorMsg = ''
+            this.stageVariables.errorMsg = ''
         },
         launch() {
-            this.resetJsonError()
-            let payloadJson: { [k: string]: any } = {}
-            if (this.payload !== '') {
-                try {
-                    payloadJson = JSON.parse(this.payload)
-                } catch (e) {
-                    this.jsonError = e
-                    console.log(e)
-                    return
-                }
-            }
-
+            this.formatDataAndExecute('invokeLaunchConfig')
+        },
+        save() {
+            this.formatDataAndExecute('saveLaunchConfig')
+        },
+        loadConfig() {
+            this.resetJsonErrors()
             vscode.postMessage({
-                command: 'invokeLaunchConfig',
-                data: {
-                    launchConfig: {
-                        ...this.launchConfig,
-                        lambda: {
-                            ...this.launchConfig.lambda,
-                            payload: {
-                                ...this.launchConfig.payload,
-                                json: payloadJson,
-                            },
-                        },
-                    },
-                },
+                command: 'loadSamLaunchConfig',
             })
         },
-        save() {
-            this.resetJsonError()
-            let payloadJson: { [k: string]: any } = {}
-            if (this.payload !== '') {
+        loadPayload() {
+            this.resetJsonErrors()
+            vscode.postMessage({
+                command: 'getSamplePayload',
+            })
+        },
+        loadResource() {
+            this.resetJsonErrors()
+            vscode.postMessage({
+                command: 'getTemplate',
+            })
+        },
+        toggleShowAllFields() {
+            this.showAllFields = !this.showAllFields
+        },
+        stringToBoolean(field: string){
+            if (field === "True") {
```

Comment:
Nit: this entire function can be shortened to `return field === 'True'`

Unsure what happens if the `else if` isn't triggered.

Reply:
refactored